src/lib/wagmi.ts

```diff
@@ -107,43 +106,53 @@ enum ChainId {
   BASE_SEPOLIA = '84532',
 }
 
-const getAlchemyRPCUrls = (chainId: ChainId, apiKey: string) => {
+const getAlchemyRPCUrlForChainId = (chainId: ChainId) => {
+  const alchemyKey = import.meta.env.VITE_ALCHEMY_API_KEY;
+  if (!alchemyKey) return undefined;
   switch (chainId) {
     case ChainId.ETH_MAINNET:
-      return `https://eth-mainnet.g.alchemy.com/v2/${apiKey}`;
+      return `https://eth-mainnet.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.ETH_SEPOLIA:
-      return `https://eth-sepolia.g.alchemy.com/v2/${apiKey}`;
+      return `https://eth-sepolia.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.POLYGON_MAINNET:
-      return `https://polygon-mainnet.g.alchemy.com/v2/${apiKey}`;
+      return `https://polygon-mainnet.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.POLYGON_MUMBAI:
-      return `https://polygon-amoy.g.alchemy.com/v2/${apiKey}`;
+      return `https://polygon-amoy.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.OPT_MAINNET:
-      return `https://opt-mainnet.g.alchemy.com/v2/${apiKey}`;
+      return `https://opt-mainnet.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.OPT_SEPOLIA:
-      return `https://opt-sepolia.g.alchemy.com/v2/${apiKey}`;
+      return `https://opt-sepolia.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.ARB_MAINNET:
-      return `https://arb-mainnet.g.alchemy.com/v2/${apiKey}`;
+      return `https://arb-mainnet.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.ARB_SEPOLIA:
-      return `https://arb-sepolia.g.alchemy.com/v2/${apiKey}`;
+      return `https://arb-sepolia.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.BASE_MAINNET:
-      return `https://base-mainnet.g.alchemy.com/v2/${apiKey}`;
+      return `https://base-mainnet.g.alchemy.com/v2/${alchemyKey}`;
     case ChainId.BASE_SEPOLIA:
-      return `https://base-sepolia.g.alchemy.com/v2/${apiKey}`;
+      return `https://base-sepolia.g.alchemy.com/v2/${alchemyKey}`;
     default:
       return undefined;
   }
 };
 
+export const RPCUrlsByChainId = [mainnet, ...WAGMI_SUPPORTED_CHAINS].reduce(
```

Comment:
i tried for a while to make a test for this, but importing privy seems to break vitest 😢 
happy to look into this at a later point but didnt' feel like it was worth the time now